cita-executor/core/src/cita_executive.rs

```diff
@@ -639,56 +651,51 @@ pub fn contract_address(address: &Address, nonce: &U256) -> Address {
     From::from(stream.out().crypt_hash())
 }
 
-#[derive(Debug)]
+// All the EVMError will be set into ExecutedResult's exception.
+// And some of them which need return Err in CitaExecutive.exec will be set to ExecutionError too.
+/// Error Result for execute a transaction
+#[derive(Debug, PartialEq)]
 pub enum ExecutionError {
-    VM(VMError),
-    State(StateError),
-    NotFound,
-    Types(TypesError),
+    /// Return when internal vm error occurs.
     Internal(String),
+    /// Return when generic transaction occurs.
     TransactionMalformed(String),
+    /// Return when authentication error occurs.
     Authentication(AuthenticationError),
-    NativeContract(NativeError),
+    /// Return when the quota_limit in transaction lower then base quota required.
+    NotEnoughBaseGas,
+    /// Return when transaction nonce does not match state nonce.
+    InvalidNonce,
+    /// Return when the cost of transaction (value + quota_price * quota) exceeds.
+    NotEnoughBalance,
+    /// Return when the block quota exceeds block quota limit.
```

Comment:
I am a little confused about what the difference between `ExecutionError` and `ExecutedException`.

`ExecutedException` wraps error returned from vm?  if this logic is right,  Maybe `InvalidNonce` and `NotEnoughBalance` belongs to `ExecutedException`.

Reply:
`ExecutedException` will be set into `Receipt`, but `ExecutionError` will be handled in `block.rs` where called the `exec` function.